solver/src/main/java/org/chocosolver/solver/variables/view/integer/IntAffineView.java

```diff
@@ -0,0 +1,588 @@
+                        min = max = vit.next();
+                    }
+
+                    @Override
+                    public void topDownInit() {
+                        vit = getValueIterator(false);
+                        iterable = vit.hasPrevious();
+                        min = max = vit.previous();
+                    }
+
+                    @Override
+                    public boolean hasNext() {
+                        boolean isIterable = iterable;
+                        iterable = vit.hasNext();
+                        return isIterable;
+                    }
+
+                    @Override
+                    public boolean hasPrevious() {
+                        boolean isIterable = iterable;
+                        iterable = vit.hasPrevious();
+                        return isIterable;
+                    }
+
+                    @Override
+                    public void next() {
+                        min = max = vit.next();
+                    }
+
+                    @Override
+                    public void previous() {
+                        min = max = vit.previous();
+                    }
+
+                    @Override
+                    public int min() {
+                        return min;
+                    }
+
+                    @Override
+                    public int max() {
+                        return max;
+                    }
+                };
+            }
+        }
+        if (bottomUp) {
+            _riterator.bottomUpInit();
+        } else {
+            _riterator.topDownInit();
+        }
+        return _riterator;
+    }
+
+    @Override
+    public boolean contains(int value) {
+        value -= b;
+        if (a > 1) {
+            if ((value % a) != 0) {
+                return false;
+            }
+            value /= a;
+        }
+        value = (!p) ? -value : value;
+        return var.contains(value);
+    }
+
+    @Override
+    public boolean isInstantiatedTo(int value) {
+        return var.isInstantiated() && contains(value);
+    }
+
+    @Override
+    public int getValue() throws IllegalStateException {
+        if (!isInstantiated()) {
+            throw new IllegalStateException("getValue() can be only called on instantiated variable. " +
+                    name + " is not instantiated");
+        }
+        return (p ? var.getValue() : -var.getValue()) * a + b;
+    }
+
+    @Override
+    public int getLB() {
+        return (!p ? -var.getUB() : var.getLB()) * a + b;
+    }
+
+    @Override
+    public int getUB() {
+        return (!p ? -var.getLB() : var.getUB()) * a + b;
+    }
+
+    @Override
+    public int nextValue(int v) {
+        if (v < getLB()) return getLB();
+        if (v > getUB()) return Integer.MAX_VALUE;
+        // y = p * a * x + b where p in {-1,1}, a and b in Z.
+        // => x = (y - b) / (p * a)
+        v -= b;
+        v = (v < 0 && a > 1) ? v - 1 : v;
```

Comment:
Should be replaced by `v = (v < 0 && a > 1 && v % a != 0) ? v - a : v;`

See below why